storage/local/persistence.go

```diff
@@ -812,6 +812,7 @@ func (p *persistence) dropAndPersistChunks(
 		}
 		_, err = io.ReadFull(f, headerBuf)
 		if err == io.EOF {
+			f.Close()
```

Comment:
Could you add a comment why this Close() is required?
I can imagine that later, developers will think it's not needed and remove it again. In particular since the `defer f.Close()` in line 803 will close the closed file again (which is OK, the 2nd Close call will return an error, but that doesn't matter, and it will not change any state - that's perhaps something to include in the comment as well).